hub/core/meta/dataset_meta.py

```diff
@@ -1,18 +1,34 @@
-from hub.constants import META_ENCODING
-import json
-
-import hub
-from hub.core.typing import StorageProvider
+from typing import Dict, List
+from hub.core.storage.provider import StorageProvider
+from hub.core.meta.meta import Meta
 from hub.util.keys import get_dataset_meta_key
 
 
-def write_dataset_meta(storage: StorageProvider, meta: dict):
-    storage[get_dataset_meta_key()] = json.dumps(meta).encode(META_ENCODING)
+class DatasetMeta(Meta):
+    tensors: List
+
+    @staticmethod
+    def create(storage: StorageProvider):
+        """Dataset metadata is responsible for keeping track of global tensor metadata and where tensors exist.
+
+        Note:
+            Dataset metadata that is automatically synchronized with `storage`. For more details, see the `Meta` class.
+            Auto-populates `required_meta` that `Meta` accepts as an argument.
+
+        Args:
+            storage (StorageProvider): Destination of this meta. No `key` argument required, the
+                dataset meta file will be added to the root of `storage`.
 
+        Returns:
+            DatasetMeta: Dataset meta object.
+        """
 
-def read_dataset_meta(storage: StorageProvider) -> dict:
-    return json.loads(storage[get_dataset_meta_key()])
+        required_meta: Dict = {
+            "tensors": [],
+        }
 
+        return DatasetMeta(get_dataset_meta_key(), storage, required_meta=required_meta)
 
-def default_dataset_meta():
-    return {"tensors": [], "version": hub.__version__}
+    @staticmethod
+    def load(storage: StorageProvider):
+        return DatasetMeta(get_dataset_meta_key(), storage)
```

Comment:
If the first parameter is the same in both cases, does it really need to be a parameter at all? After removing that, I think you'll find that `.load` and `.__init__` are exactly the same.

Reply:
this is more for uniformity because `TensorMeta` has extra  arguments beforehand

Reply:
and also, `TensorMeta.create(...)` tells me more information than `TensorMeta(...)`, it feels easier to read

Reply:
we can change this later during refactor and really find the best way, im open to having my mind changed

Reply:
@McCrearyD `.create` absolutely says more than `.__init__`. But `.load` doesn't. I still think it should just alias the parent constructor you appear to be calling here.